hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/mor/MergeOnReadTableState.java

```diff
@@ -83,11 +96,20 @@ public int getOperationPos() {
   }
 
   public int[] getRequiredPositions() {
-    final List<String> fieldNames = rowType.getFieldNames();
```

Comment:
Can we add the table name param for this method `getRequiredPositions` instead?

Reply:
The method getRequiredPositions is called by objects CdcInputFormat and MergeOnReadInputFormat，neither of these two objects stores table names.
![image](https://github.com/apache/hudi/assets/34104400/4981da57-81e7-4c3c-92a0-a10f48c4e0bb)

Reply:
Just fetch it with `OptionsResolver` and param `conf`.

Reply:
done.